scripts/check_error_codes.py

```diff
@@ -102,38 +102,100 @@ def fix_ids(used_ids, file_names):
         fix_ids_in_file(file_name, available_ids, used_ids)
 
 
-def find_source_files(top_dir):
-    """Builds the list of .h and .cpp files in top_dir directory"""
+def find_files(top_dir, sub_dirs, extensions):
+    """Builds a list of files with given extensions in specified subdirectories"""
 
     source_file_names = []
-    dirs = ['libevmasm', 'liblangutil', 'libsolc', 'libsolidity', 'libsolutil', 'libyul', 'solc']
-
-    for dir in dirs:
+    for dir in sub_dirs:
         for root, _, file_names in os.walk(os.path.join(top_dir, dir), onerror=lambda e: exit(f"Walk error: {e}")):
             for file_name in file_names:
                 _, ext = path.splitext(file_name)
-                if ext in [".h", ".cpp"]:
+                if ext in extensions:
                     source_file_names.append(path.join(root, file_name))
 
     return source_file_names
 
 
+def find_ids_in_test_file(file_name, ids):
+    source = read_file(file_name)
+    pattern = r"^// (.*Error|Warning) \d\d\d\d:"
+    for m in re.finditer(pattern, source, flags=re.MULTILINE):
+        id = m.group(0)[-5:-1]
+        ids.add(id)
+
+
+def find_ids_in_test_files(file_names):
+    used_ids = set()
+    for file_name in file_names:
+        find_ids_in_test_file(file_name, used_ids)
+    return used_ids
```

Comment:
```suggestion
def find_ids_in_test_file(file_name):
    source = read_file(file_name)
    pattern = r"^// (.*Error|Warning) \d\d\d\d:"
    return (m.group(0)[-5:1] for m in re.finditer(pattern, source, flags=re.MULTILINE))

def find_ids_in_test_files(file_names):
    return set(find_ids_in_test_file(file_name) for file_name in file_names)
```

Using comprehensions might make it less readable for people not used to python, but I don't see a need for an in-out-parameter at all.

Reply:
Would `set(find_ids_in_test_file ....)` create a union? Do not we get a set of generators instead?

Comprehensions are fine and should be encouraged. And if these two functions would be taken out of context, I would do something pretty similar to your suggestion, including merging them into one.

My reasons to keep existing code are:
* it is good to have the same scan pattern, `find_all_ids` + `find_ids_in_file`, etc., for both source files and tests;
* we may change `set` to `dict`, as for source files; and it is harder to union dicts than sets.

Anyway, these are just a couple of lines. So if you insist, we can go for your variant.

Reply:
Please either remove the in-out parameter or document it.

Reply:
I changed to

```
def find_ids_in_test_file(file_name):
    source = read_file(file_name)
    pattern = r"^// (.*Error|Warning) \d\d\d\d:"
    return {m.group(0)[-5:-1] for m in re.finditer(pattern, source, flags=re.MULTILINE)}

def find_ids_in_test_files(file_names):
    used_ids = set()
    for file_name in file_names:
        used_ids |= find_ids_in_test_file(file_name)
    return used_ids
```